x-pack/plugins/security_solution/public/detections/components/alerts_table/timeline_actions/alert_context_menu.tsx

```diff
@@ -0,0 +1,446 @@
+        alertStatus,
+      ]
+    );
+
+    const button = (
+      <EuiButtonIcon
+        aria-label="context menu"
+        data-test-subj="timeline-context-menu-button"
+        size="s"
+        iconType="boxesHorizontal"
+        onClick={onButtonClick}
+      />
+    );
+
+    const handleAddEndpointExceptionClick = useCallback(() => {
+      const [ruleName] = getMappedNonEcsValue({
+        data: nonEcsRowData,
+        fieldName: 'signal.rule.name',
+      });
+      const [ruleId] = getMappedNonEcsValue({
+        data: nonEcsRowData,
+        fieldName: 'signal.rule.id',
+      });
+      const ruleIndices = getMappedNonEcsValue({
+        data: nonEcsRowData,
+        fieldName: 'signal.rule.index',
+      });
+
+      closePopover();
+
+      if (ruleId !== undefined) {
+        openAddExceptionModal({
+          ruleName: ruleName ?? '',
+          ruleId,
+          ruleIndices: ruleIndices.length > 0 ? ruleIndices : DEFAULT_INDEX_PATTERN,
+          exceptionListType: 'endpoint',
+          alertData: {
+            ecsData: ecsRowData,
+            nonEcsData: nonEcsRowData,
+          },
+        });
+      }
+    }, [closePopover, ecsRowData, nonEcsRowData, openAddExceptionModal]);
+
+    const addEndpointExceptionComponent = (
+      <EuiContextMenuItem
+        key="add-endpoint-exception-menu-item"
+        aria-label="Add Endpoint Exception"
+        data-test-subj="add-endpoint-exception-menu-item"
+        id="addEndpointException"
+        onClick={handleAddEndpointExceptionClick}
+        disabled={!canUserCRUD || !hasIndexWrite || !isEndpointAlert}
+      >
+        <EuiText size="m">{i18n.ACTION_ADD_ENDPOINT_EXCEPTION}</EuiText>
+      </EuiContextMenuItem>
+    );
+
+    const handleAddExceptionClick = useCallback(() => {
+      const [ruleName] = getMappedNonEcsValue({
+        data: nonEcsRowData,
+        fieldName: 'signal.rule.name',
+      });
+      const [ruleId] = getMappedNonEcsValue({
+        data: nonEcsRowData,
+        fieldName: 'signal.rule.id',
+      });
+      const ruleIndices = getMappedNonEcsValue({
+        data: nonEcsRowData,
+        fieldName: 'signal.rule.index',
+      });
+
+      closePopover();
+
+      if (ruleId !== undefined) {
+        openAddExceptionModal({
+          ruleName: ruleName ?? '',
+          ruleId,
+          ruleIndices: ruleIndices.length > 0 ? ruleIndices : DEFAULT_INDEX_PATTERN,
+          exceptionListType: 'detection',
+          alertData: {
+            ecsData: ecsRowData,
+            nonEcsData: nonEcsRowData,
+          },
+        });
+      }
+    }, [closePopover, ecsRowData, nonEcsRowData, openAddExceptionModal]);
+
+    const addExceptionComponent = (
+      <EuiContextMenuItem
+        key="add-exception-menu-item"
+        aria-label="Add Exception"
+        data-test-subj="add-exception-menu-item"
+        id="addException"
+        onClick={handleAddExceptionClick}
+        disabled={!canUserCRUD || !hasIndexWrite}
```

Comment:
#75802 added a small bit of disabling logic to the original `default_config.tsx` file that would fit in here. Believe that's now causing the merge conflict